From: Ben Hutchings <ben@decadent.org.uk>
Subject: [PATCH] mm/mremap: Fix conflict between 2.6.32.4 and vserver

Make some cosmetic changes to the reworked mremap in 2.6.32.4 so that
the vserver patch will still apply (with fuzz 1).

--- a/mm/mremap.c
+++ b/mm/mremap.c
@@ -288,11 +288,11 @@
 		lock_limit = current->signal->rlim[RLIMIT_MEMLOCK].rlim_cur;
 		locked += new_len - old_len;
 		if (locked > lock_limit && !capable(CAP_IPC_LOCK))
-			goto Eagain;
+			goto out;
 	}
-
-	if (!may_expand_vm(mm, (new_len - old_len) >> PAGE_SHIFT))
+	if (!may_expand_vm(mm, (new_len - old_len) >> PAGE_SHIFT)) {
 		goto Enomem;
+	}
 
 	if (vma->vm_flags & VM_ACCOUNT) {
 		unsigned long charged = (new_len - old_len) >> PAGE_SHIFT;
@@ -309,7 +309,7 @@
 	return ERR_PTR(-EINVAL);
 Enomem:
 	return ERR_PTR(-ENOMEM);
-Eagain:
+out:
 	return ERR_PTR(-EAGAIN);
 }
 
